book/listings/gobject_properties/3/custom_button/imp.rs

```diff
@@ -0,0 +1,83 @@
+use glib::{BindingFlags, Object};
+use glib::{ParamFlags, ParamSpec, Value};
+use gtk::subclass::prelude::*;
+use gtk::{glib, Align, Orientation};
+use gtk::{prelude::*, BoxBuilder};
+use gtk::{Application, ApplicationWindowBuilder};
+use once_cell::sync::Lazy;
+use std::cell::RefCell;
+
+// Object holding the state
+#[derive(Default)]
+pub struct CustomButton {
+    number: RefCell<i32>,
+}
+
+// The central trait for subclassing a GObject
+#[glib::object_subclass]
+impl ObjectSubclass for CustomButton {
+    const NAME: &'static str = "MyGtkAppCustomButton";
+    type Type = super::CustomButton;
+    type ParentType = gtk::Button;
+}
+// ANCHOR: object_impl
+// Trait shared by all GObjects
+impl ObjectImpl for CustomButton {
+    fn constructed(&self, obj: &Self::Type) {
+        self.parent_constructed(obj);
+        obj.set_label(&self.number.borrow().to_string());
+    }
+
+    fn properties() -> &'static [ParamSpec] {
+        static PROPERTIES: Lazy<Vec<ParamSpec>> = Lazy::new(|| {
+            vec![ParamSpec::int(
+                // Name
+                "number",
+                // Nickname
+                "number",
+                // Short description
+                "number",
+                // Minimum value
+                i32::MIN,
+                // Maximum value
+                i32::MAX,
+                // Default value
+                0,
+                // The property can be read and written to
+                ParamFlags::READWRITE,
+            )]
+        });
+        PROPERTIES.as_ref()
+    }
+    fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
```

Comment:
Could use an empty line here to be consistent with the other functions in ObjectImpl